core/src/shred.rs

```diff
@@ -26,14 +30,133 @@ lazy_static! {
         { bincode::serialized_size(&Signature::default()).unwrap() as usize };
     static ref SIZE_OF_EMPTY_VEC: usize =
         { bincode::serialized_size(&vec![0u8; 0]).unwrap() as usize };
+    static ref SIZE_OF_SHRED_TYPE: usize = { bincode::serialized_size(&0u8).unwrap() as usize };
 }
 
-#[derive(Serialize, Clone, Deserialize, PartialEq, Debug)]
-pub struct ShredMetaBuf {
-    pub slot: u64,
-    pub index: u32,
-    pub data_shred: bool,
-    pub shred_buf: Vec<u8>,
+/// The constants that define if a shred is data or coding
+const DATA_SHRED: u8 = 0b10100101;
+const CODING_SHRED: u8 = 0b01011010;
+
+#[derive(Clone, Debug)]
+pub struct ShredInfo {
+    pub headers: DataShredHeader,
+    pub shred: Vec<u8>,
+}
+
+impl ShredInfo {
+    fn new(header: DataShredHeader, shred_buf: Vec<u8>) -> Self {
+        ShredInfo {
+            headers: header,
+            shred: shred_buf,
+        }
+    }
+
+    pub fn new_from_serialized_shred(shred_buf: Vec<u8>) -> Self {
+        let mut header = DataShredHeader::default();
```

Comment:
probably really minor but you can avoid this default construction. 
Just leave it uninitialized here and construct it field by field in the Coding shred case and just directly assign in the else.